acvm-repo/bn254_blackbox_solver/src/schnorr/mod.rs

```diff
@@ -0,0 +1,146 @@
+use acvm_blackbox_solver::blake2s;
+use ark_ec::{
+    short_weierstrass::{Affine, SWCurveConfig},
+    AffineRepr, CurveConfig, CurveGroup,
+};
+use ark_ff::{BigInteger, PrimeField, Zero};
+use grumpkin::{Fq, GrumpkinParameters};
+
+pub(crate) fn verify_signature(
+    pub_key_x: Fq,
+    pub_key_y: Fq,
+    sig_s_bytes: [u8; 32],
+    sig_e_bytes: [u8; 32],
+    message: &[u8],
+) -> bool {
+    let pub_key = Affine::<GrumpkinParameters>::new_unchecked(pub_key_x, pub_key_y);
+
+    if !pub_key.is_on_curve()
+        || !pub_key.is_in_correct_subgroup_assuming_on_curve()
+        || pub_key.is_zero()
+    {
+        return false;
+    }
+
+    let sig_s =
+        <GrumpkinParameters as CurveConfig>::ScalarField::from_be_bytes_mod_order(&sig_s_bytes);
+    let sig_e =
+        <GrumpkinParameters as CurveConfig>::ScalarField::from_be_bytes_mod_order(&sig_e_bytes);
+
+    if sig_s.is_zero() || sig_e.is_zero() {
+        return false;
+    }
+
+    // R = g^{sig.s} • pub^{sig.e}
+    let r = pub_key * sig_e + GrumpkinParameters::GENERATOR * sig_s;
```

Comment:
```suggestion
    let r = GrumpkinParameters::GENERATOR * sig_s + pub_key * sig_e;
```
Nit: but do we want to switch either the order of this equation or the comment

Reply:
This matches wikipedia's ordering so lets switch the code.